snuba_sdk/request.py

```diff
@@ -82,6 +83,14 @@ def to_dict(self) -> dict[str, object]:
             assert isinstance(serialized_mql, dict)  # mypy
             mql_context = serialized_mql["mql_context"]
             query = str(serialized_mql["mql"])
+        elif isinstance(self.query, DeleteQuery):
+            return {
+                **flags,
+                "query": self.query.serialize(),
+                "app_id": self.app_id,
+                "tenant_ids": self.tenant_ids,
+                "parent_api": self.parent_api,
+            }
```

Comment:
obviously this is a duct tape solution. Its what we will use for now since we kind of have to use the Request object right now.

A longer term solution would likely be to refactor this request object. Maybe considered down the line.